libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialog.tsx

```diff
@@ -0,0 +1,39 @@
+// (C) 2021 GoodData Corporation
+import React, { useCallback, useMemo } from "react";
+import { ShareDialogBase } from "./ShareDialogBase/ShareDialogBase";
+import { GranteeItem } from "./ShareDialogBase/types";
+import { IShareDialogProps } from "./types";
+import invariant from "ts-invariant";
+import { mapGranteesToShareStatus, mapOwnerToGrantee, mapShareStatusToGroupAll } from "./shareDialogMappers";
+
+/**
+ * @internal
+ */
+export const ShareDialog: React.FC<IShareDialogProps> = (props) => {
+    const { sharedObject, currentUserRef, onApply, onCancel } = props;
+
+    const owner = useMemo(() => {
+        invariant(sharedObject.createdBy, "ShareDialog sharedObject.createdBy should be specified.");
+
+        return mapOwnerToGrantee(sharedObject.createdBy, currentUserRef);
+    }, [sharedObject, currentUserRef]);
+
+    const grantees = useMemo(() => {
+        const groupAll = mapShareStatusToGroupAll(sharedObject.shareStatus);
+        if (groupAll) {
+            return [groupAll];
+        }
+        return [];
+    }, [sharedObject]);
```

Comment:
not expert, however isn't it safer to memo on the actual value instead of the entire object? e.g. extract createdBy & sharedStatus and use those directly & as memo deps?

Reply:
Done